src/toil/wdl/wdltoil.py

```diff
@@ -89,6 +98,14 @@
 
 logger = logging.getLogger(__name__)
 
+# WDL options to pass into the WDL jobs and standard libraries
+#   task_path: Dotted WDL name of the part of the workflow this library is working for.
+#   execution_dir: Directory to use as the working directory for workflow code.
+#   enforce_existence: If true, then if a file is detected as nonexistent, raise an error. Else, let it pass through
+#   share_files_with: If set to an existing standard library instance, use the same file upload and download paths as it.
+WDL_OPTIONS = TypedDict('WDL_OPTIONS', {"execution_dir": NotRequired[str], "container": NotRequired[str],
+                                        "share_files_with": NotRequired["ToilWDLStdLibBase"], "task_path": str})
+
```

Comment:
I don't think the design here is right.

If we want this to be "options", it should contain *only* things that are global constants across the workflow execution, which here would be the `execution_dir` (leader working directory) and the `container` (which container runner to use).

I think it does make sense to have it also carry `task_path` so we don't need to keep passing the task path around through everything and manually giving it to the standard library. But then we should probably call this a `wdl_context` rather than `wdl_options`, and it would have a tree structure: you derive a child context that is more specific from a parent context that is more general, and you use it for part of the workflow tree, mirroring the dotted task name structure. Also in that case we would also want to carry `namespace` alongside `task_path`.

I think it makes no sense at all to have either `enforce_existence` or `share_files_with` as part of this. They don't apply to a whole workflow or to a subtree of a workflow; if you actually passed one of them into a job from above there'd be no way to sensibly respect it. They should go back to being separate arguments on the standard library/other file access functions.

If we make those changes, then we can have a nice controlled way in which we work with the options/context information. We always either pass it straight through, or make a copy of it and adjust it to make the `task_path`/`namespace` more specific. We only ever create it once, on the leader. Then we never have to worry about if we are looking at the real options/context passed down from the top or one that was made as a dict literal in a function call.

Reply:
I removed `enforce_existence` and `share_files_with` away from the WDL options so they should be more global now